webdriver/tests/bidi/script/call_function/call_function.py

```diff
@@ -0,0 +1,162 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget, ScriptEvaluateResultException
+from ... import recursive_compare, any_string, any_int
+from .. import any_stack_trace
+
+
+@pytest.mark.asyncio
+async def test_exception(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration='()=>{throw 1}',
+            await_promise=False,
+            target=ContextTarget(top_context["context"]))
+
+    recursive_compare({
+        'realm': any_string,
+        'exceptionDetails': {
+            'columnNumber': any_int,
+            'exception': {
+                'value': 1,
+                'type': 'number'},
+            'lineNumber': any_int,
+            'stackTrace': any_stack_trace,
+            'text': any_string}},
+        exception.value.result)
+
+
+@pytest.mark.asyncio
+async def test_invalid_function(bidi_session, top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration='))) !!@@## some invalid JS script (((',
+            await_promise=False,
+            target=ContextTarget(top_context["context"]))
+    recursive_compare({
+        'realm': any_string,
+        'exceptionDetails': {
+            'columnNumber': any_int,
+            'exception': {
+                'handle': any_string,
+                'type': 'error'},
+            'lineNumber': any_int,
+            'stackTrace': any_stack_trace,
+            'text': any_string}},
+        exception.value.result)
+
+
+@pytest.mark.asyncio
+async def test_arrow_function(bidi_session, top_context):
+    result = await bidi_session.script.call_function(
+        function_declaration="()=>{return 1+2;}",
+        await_promise=False,
+        target=ContextTarget(top_context["context"]))
+
+    assert result == {
+        "type": "number",
+        "value": 3}
+
+
+@pytest.mark.asyncio
+async def test_arguments(bidi_session, top_context):
```

Comment:
It would be interesting to have variants of both `test_arguments` and `test_this` using the default values of `arguments` and `this`, respectively. For arguments, we should expect to get an empty array. But for a `return this` when `this` is not provided, I'm not sure what we should expect. Per spec the this object should be `null` but this can either mean we fallback to the global in non-strict mode, or we actually use null in strict mode.

So maybe add a test for the default `arguments`, and we can discuss what should happen for a default `this` in a follow up. FYI, filed an issue about strict mode at https://github.com/w3c/webdriver-bidi/issues/251.

Reply:
Added test fulfilling the spec and falling back `null` as `this` to `window` object.